baseplate/sidecars/live_data_watcher.py

```diff
@@ -32,6 +37,64 @@ def __init__(self, dest: str, owner: int, group: int, mode: int):
         self.group = group
         self.mode = mode
 
+    @staticmethod
+    def get_encrypted_data_from_s3(
+        bucket_name: str, file_key: str, region_name: str, sse_key: str
+    ) -> Optional[bytes]:
+        s3_client = boto3.client(
+            "s3",
+            region_name=region_name,
+        )
+        data = None
+        # The S3 data must be Server Side Encrypted and we should have the decryption key.
+        kwargs = {
+            "Bucket": bucket_name,
+            "Key": file_key,
+            "SSECustomerKey": sse_key,
+            "SSECustomerAlgorithm": "AES256",
+        }
+        try:
+            s3_object = s3_client.get_object(**kwargs)
+            # Returns bytes.
+            data = s3_object["Body"].read()
+        except ClientError as error:
+            logger.exception(
+                "Failed to load from S3. Received error code %s: %s",
+                error.response["Error"]["Code"],
+                error.response["Error"]["Message"],
+            )
+            return None
```

Comment:
Why does this log the error and return none, rather than raising an error?

Reply:
I felt that logging and error and returning None would be a more graceful way to handle this in case something went wrong. Wouldn't raising an error be a more nuclear option in this case? It seemed valid to me that if this S3 flow fails, just return None back to the on_change() function so that we don't completely crash out